File: Week_02/kotlin/src/_404_SumOfLeftLeaves.kt
import java.util.*

class _404_UsingDfsIteration {
    fun sumOfLeftLeaves(root: TreeNode?): Int {
        var node = root
        val stack = LinkedList<TreeNode>()
        var ret = 0
        while (node != null || stack.isNotEmpty()) {
            while (node != null) {
                stack.push(node)
                val left = node.left
                if (left != null && left.left == null && left.right == null) {
                    ret += left.`val`
                    node = null
                } else {
                    node = node.left
                }
            }
            node = stack.pop().right
        }
        return ret
    }
}

File: Week_02/c/src/_404_SumOfLeftLeaves.cpp
struct TreeNode
{
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
};

int sumOfLeftLeaves(struct TreeNode *root)
{
    if (!root)
        return 0;
    int ret = 0;
    struct TreeNode *l = root->left;
    if (l)
    {
        if (!l->left && !l->right)
        {
            ret += l->val;
        }
        else
        {
            ret += sumOfLeftLeaves(l);
        }
    }
    return ret + sumOfLeftLeaves(root->right);
}